source/common/upstream/conn_pool_map_impl.h

```diff
@@ -52,14 +52,28 @@ ConnPoolMap<KEY_TYPE, POOL_TYPE>::getPool(KEY_TYPE key, const PoolFactory& facto
   // We have room for a new pool. Allocate one and let it know about any cached callbacks.
   auto new_pool = factory();
   connPoolResource.inc();
-  for (const auto& cb : cached_callbacks_) {
-    new_pool->addDrainedCallback(cb);
+  for (const auto& [cb, drain] : cached_callbacks_) {
+    new_pool->addIdleCallback(cb, drain);
   }
```

Comment:
It seems a bit odd to be able to immediately trigger a drain? Is this possible in practice or just a side effect of the new API?

Reply:
I think this will only happen if `drain == DrainPool::No`, meaning it must be an idle callback, which will not trigger draining. I'll add an ASSERT so it's clearer what is going on here.